drivers/sensor/st/lsm6dso/lsm6dso_trigger.c

```diff
@@ -84,6 +85,89 @@ static int lsm6dso_enable_xl_int(const struct device *dev, int enable)
 	}
 }
 
+/**
+ * lsm6dso_enable_xl_motion_int - XL enable selected int pin to generate
+ * interrupt when any motion is detected.
+ */
+static int lsm6dso_enable_xl_delta_int(const struct device *dev, int enable)
+{
+	const struct lsm6dso_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	int rc;
+
+	/* Only configure the tap registers when we're enabling, as we'll
+	 * otherwise get a spurious interrupt when this returns!
+	 */
+	if (enable) {
+		lsm6dso_tap_cfg0_t tap_cfg0;
+		lsm6dso_tap_cfg2_t tap_cfg2;
+
+		rc = lsm6dso_read_reg(ctx, LSM6DSO_TAP_CFG0, (uint8_t *)&tap_cfg0, 1);
+		if (rc < 0) {
+			return rc;
+		}
+
+		/* Output state changes, and use the slope filter. */
+		tap_cfg0.sleep_status_on_int = 0;
+		tap_cfg0.slope_fds = 0;
+
+		rc = lsm6dso_write_reg(ctx, LSM6DSO_TAP_CFG0, (uint8_t *)&tap_cfg0, 1);
+		if (rc < 0) {
+			return rc;
+		}
```

Comment:
This is technically fine but you should really verify that the new value is different from what's already there. An easier way might be to leverage `i2c_reg_update_byte` and `i2c_reg_update_byte_dt` from `<zephyr/drivers/i2c.h>`.

Reply:
I had considered dropping the read-modify-write here - but the values in this register also control latching and clearing behaviour.

`i2c_reg_update_byte` looks like a much better means to set these values - I will make this change :)

Reply:
On second thought, that makes it harder to use the `lsm6dso_read_reg` and `lsm6dso_write_reg` functions, which is what we're using to abstract I2C and SPI.

There is a built-in `lsm6dso_act_pin_notification_set` function - which sets/clears the `tap_cfg0.sleep_status_on_int` bit - and a `lsm6dso_xl_hp_path_internal_set` function. However this seems less efficient to me (two reads and two writes, vs current one read one write).

I like your idea, but it goes against the grain a bit. Keen to hear your thoughts.

Reply:
I'm fine to leave this as is. Just wanted to make sure you knew and considered using the `update_byte` functions

Reply:
Thanks for pointing it out! I'll consider this as I push up my other drivers.